src/package/script.rs

```diff
@@ -133,12 +133,15 @@ impl<'a> HighlightedScript<'a> {
 
         let mut h = HighlightLines::new(syntax, theme);
 
-        Ok({
-            LinesWithEndings::from(&self.script.0).map(move |line| {
-                let ranges: Vec<(Style, &str)> = h.highlight(line, &self.ps);
-                as_24_bit_terminal_escaped(&ranges[..], true)
-            })
-        })
+        let mut highlighted_lines = Vec::new();
+        for line in LinesWithEndings::from(&self.script.0) {
+            let ranges: Vec<(Style, &str)> = h
+                .highlight_line(line, &self.ps)
+                .with_context(|| anyhow!("Could not highlight the following line: {}", line))?;
+            highlighted_lines.push(as_24_bit_terminal_escaped(&ranges[..], true));
+        }
+
+        Ok(highlighted_lines.into_iter())
```

Comment:
I just rebased this PR onto `master` and added this change to fix the CI error. `highlight_line` now returns a `Result` instead of causing panics. We could either use `unwrap`/`expect`, etc. to keep the old behavior or handle the error in a nicer way. I decided to go with the latter approach since the `lines` function already returns a `Result`. Dealing with error handling on iterators was a bit too complex for me and wasn't any prettier IMO (I found solutions using `scan()`, `collect()` (very similar to my current approach), or using a `Result<String>` as item) so I decided to switch to a for loop instead to make it more beginner friendly :D I expect it to be a bit less efficient due to the `Iterator -> Vec -> Iterator` conversions but it shouldn't really matter at all.

It isn't that pretty but given the constraints it seemed to suck the least.

@matthiasbeyer is this acceptable or is there a much better solution (that isn't too complicated :D) that I missed?

Reply:
You're close!

The piece of code you pasted above looks _almost_ nice. The one thing you missed there: You collect into a `Result<Vec<_>>`. Then you `?` on that, turning it into an iterator again and wrap the whole thing with a `Ok(_)`.

Why not simply:

```rust
LinesWithEndings::from(&self.script.0)
    .map(move |line| -> Result<String> {
        let ranges: Vec<(Style, &str)> = h.highlight_line(line, &self.ps)?;
        Ok(as_24_bit_terminal_escaped(&ranges[..], true))
    })
    .collect::<Result<Vec<String>>>()
    .map(|v| v.into_iter())
```

And inside the big `map` you probably can simplify the code to:

```rust
h.highlight_line(line, &self.ps)
    .map(|r| as_24_bit_terminal_escaped(&r[..], true))
```

Does that work?

Reply:
Oh, wow, TIL that `map()` is also implemented for the `Result` type :D
(I was a bit confused until I figured that out / looked at https://doc.rust-lang.org/std/result/enum.Result.html#method.map - I still have so much to learn... :o But I guess that shouldn't be surprising given that I only just started and barely know anything about Rust yet - it'll just take me much longer than I anticipated to get a rough overview of the standard library, syntax, etc.)

Anyway, it works and makes sense :) Thanks! I also added you as co-author to the commit ;)